### ISSUE TITLE
Is it possible to include word-level confidence?

### ISSUE THREAD

**Issue by HalukMaestra** [m-bain/whisperX#141]
First of all I would like to thank you for all the great work done with this application. It's a joy to use and way better then the Vanilla whisper. One question I have is, is it possible to include word-level confidence scores inside result_aligned["word_segments"]? Obtaining ["segments"] and parsing it to get to the score is a tedious process for me since I have no need for other data expect the ones in word_segments? I was just wondering if this is viable to do?

**Comment by cristiantg** [comment]
Yes plesae, it would be nice. In a similar way as in whisper_timestamped.

**Comment by GBurg** [comment]
ok, I did some digging, and know where to find it, but it needs some serious work to make it happen.

First, initialize the model with word_timestamps=True (without_timestamps = false I added just to be sure)

`whisperx.load_model(model, device, language=lang, asr_options={"word_timestamps": True, "without_timestamps": False,})`


Then in the **whisperx/asr.py**, there is a method called **generate_segment_batched()**, which will have the new option available, passed on in its arguments, but it is not used in the function implementation. So here the new functionality should be implemented to calculate the word level scores. 

Implementation inspiration can be obtained from faster_whispers implementation, found in the **faster_whisper/transcribe.py** file in the **generate_segments()** methods, that can be found here: https://github.com/guillaumekln/faster-whisper/blob/fee52c922904d18a143faa9d89f3b42caf029e9e/faster_whisper/transcribe.py#L351

Now, it is not super straightforward, and for me not worth the time effort, but maybe someone else can get it to work and save us all.
